src/topology/metric_space/basic.lean

```diff
@@ -914,6 +936,37 @@ end
 lemma dist_pi_def (f g : Πb, π b) :
   dist f g = (sup univ (λb, nndist (f b) (g b)) : nnreal) := rfl
 
+lemma dist_pi_lt_iff {f g : Πb, π b} {r : ℝ} (hr : 0 < r) :
+  dist f g < r ↔ ∀b, dist (f b) (g b) < r :=
+begin
+  let r' : nnreal := ⟨r, le_of_lt hr⟩,
```

Comment:
@fpvandoorn recently added the `lift` tactic to help clean up arguments like this one. It becomes

```lean
begin
  lift r to nnreal using le_of_lt hr,
  rw [dist_pi_def, ←nnreal.coe_lt, finset.sup_lt_iff],
  { simp [nndist], refl },
  { assumption }
end
```
if we put the following instance somewhere above:
```lean
instance : can_lift ℝ nnreal :=
{ coe := coe,
  cond := λ r, r ≥ 0,
  prf := λ x hx, ⟨⟨x, hx⟩, rfl⟩ }
```
The instance probably belongs near the definition of `nnreal`.

Reply:
Also, I think `nnreal.coe_lt` and the related lemmas are stated backward. Their current direction is adding coercions left-to-right. This is even better:
```lean
@[elim_cast] protected lemma coe_lt'' {r₁ r₂ : nnreal} : (r₁ : ℝ) < r₂ ↔ r₁ < r₂ := iff.rfl

lemma dist_pi_lt_iff {f g : Πb, π b} {r : ℝ} (hr : 0 < r) :
  dist f g < r ↔ ∀b, dist (f b) (g b) < r :=
begin
  lift r to nnreal using le_of_lt hr,
  rw_mod_cast [dist_pi_def, finset.sup_lt_iff],
  { simp [nndist], refl },
  { assumption }
end
```